zipline/utils/calendars/exchange_calendar.py

```diff
@@ -0,0 +1,497 @@
+
+        Parameters
+        ----------
+        date : Timestamp
+            The UTC-canonicalized date whose open and close are needed.
+
+        Returns
+        -------
+        (Timestamp, Timestamp)
+            The open and close for the given date.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def session_date(self, dt):
+        """
+        Given a time, returns the UTC-canonicalized date of the exchange
+        session in which the time belongs. If the time is not in an exchange
+        session (while the market is closed), returns the date of the next
+        exchange session after the time.
+
+        Parameters
+        ----------
+        dt : Timestamp
+
+        Returns
+        -------
+        Timestamp
+            The date of the exchange session in which dt belongs.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def minutes_for_date(self, date):
+        """
+        Given a UTC-canonicalized date, returns a DatetimeIndex of all trading
+        minutes in the exchange session for that date.
+
+        SD: Sounds like @date can be an arbitrary datetime, and that we should
+        first map to an exchange session by calling self.session_date. Need to
+        check what the consumers expect.
+
+        Parameters
+        ----------
+        date : Timestamp
+            The UTC-canonicalized date whose minutes are needed.
+
+        Returns
+        -------
+        DatetimeIndex
+            A DatetimeIndex populated with all of the minutes in the
+            given date.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def minute_window(self, start, count, step=1):
+        """
+        Return a DatetimeIndex containing `count` market minutes, starting with
+        `start` and continuing `step` minutes at a time.
+
+        Parameters
+        ----------
+        start : Timestamp
+            The start of the window.
+        count : int
+            The number of minutes needed.
+        step : int
+            The step size by which to increment.
+
+        Returns
+        -------
+        DatetimeIndex
+            A window with @count minutes, starting with @start a returning
+            every @step minute.
+        """
+        raise NotImplementedError()
+
+
+_static_calendars = {}
+
+_lazy_calendar_names = ['NYSE']
+
+
+def get_calendar(name):
+    """
+    Retrieves an instance of an ExchangeCalendar whose name is given.
+
+    Parameters
+    ----------
+    name : str
+        The name of the ExchangeCalendar to be retrieved.
+    """
+    # First, check if the calendar is already registered
+    if name not in _static_calendars:
+        # The calendar is not registered, so check if it is a lazy calendar
+        if name not in _lazy_calendar_names:
+            # It's not a lazy calendar, so raise an exception
+            raise InvalidCalendarName(calendar_name=name)
+
+        if name is 'NYSE':
```

Comment:
Please don't use `is` for string comparisons.  This is effectively a random number generator.